pkg/image/common/operator.go

```diff
@@ -367,3 +372,45 @@ func (vmio *vmiOperator) Imported(old *harvesterv1.VirtualMachineImage, msg stri
 func (vmio *vmiOperator) BackingImageMissing(old *harvesterv1.VirtualMachineImage, err error) (*harvesterv1.VirtualMachineImage, error) {
 	return vmio.stateTransit(old, VMImageStateBackingImageMissing, fmt.Sprintf("Failed to get backing image: %s", err.Error()), 0, 0, 0)
 }
+
+func (vmio *vmiOperator) updateWithRetry(old *harvesterv1.VirtualMachineImage, updateFunc func(*harvesterv1.VirtualMachineImage) (*harvesterv1.VirtualMachineImage, error)) (*harvesterv1.VirtualMachineImage, error) {
+	namespace := old.Namespace
+	name := old.Name
+	var updatedVMI *harvesterv1.VirtualMachineImage
+
+	backoff := wait.Backoff{
+		Steps:    5,
+		Duration: 1 * time.Second,
+		Factor:   2.0,
+		Jitter:   0.1,
+	}
+
+	err := retry.OnError(backoff, func(err error) bool {
+		// Retry on conflict errors or transient API errors
+		return apierrors.IsConflict(err) || apierrors.IsServerTimeout(err) || apierrors.IsTimeout(err) || apierrors.IsServiceUnavailable(err)
+	}, func() error {
+		// Fetch the latest VMImage before each retry attempt to get the current resource version
+		latestVMI, err := vmio.cache.Get(namespace, name)
+		if err != nil {
+			logrus.Warnf("Failed to get latest VMImage %s/%s: %v", namespace, name, err)
+			return err
+		}
+
+		newVMI, err := updateFunc(latestVMI.DeepCopy())
+		if err != nil {
+			return err
+		}
+
+		updatedVMI, err = vmio.UpdateVMI(latestVMI, newVMI)
+		if err != nil {
+			logrus.Warnf("Failed to update VMImage size: %v, will retry if retriable", err)
```

Comment:
It looks like this might be the general retry function for any VM Image update. 
So maybe the logger should not narrow down to the `Failed to update VMImage size`.
Could be something like `Failed to update VM Image again in retry mechanism: %v, will retry if retriable`.
WDYT?